src/wallet.ts

```diff
@@ -0,0 +1,2170 @@
+            'Private view key:\n' +
+            `${privateViewKey}\n\n` +
+            'Private spend key:\n' +
+            `${privateSpendKey}\n\n` +
+            'Mnemonic seed:\n' +
+            `${mnenomicSeed}`
+            
+        clipboardy.writeSync(backupMessage);
+        notifyUser(notificationText, '{bold}{red-fg}PRIVATE KEYS COPIED TO CLIPBOARD{/}', 2000);
+        settingsWindow.focus();
+    });
+
+        // on addressbutton click
+        backupKeys.on('press', async function() {
+            const [privateSpendKey, privateViewKey] = wallet.getPrimaryAddressPrivateKeys();
+            const [mnenomicSeed] = wallet.getMnemonicSeed();
+            const backupMessage = 
+                'Public address:\n' +
+                `${addressString}\n\n` +
+                'Private view key:\n' +
+                `${privateViewKey}\n\n` +
+                'Private spend key:\n' +
+                `${privateSpendKey}\n\n` +
+                'Mnemonic seed:\n' +
+                `${mnenomicSeed}`
+                
+            clipboardy.writeSync(backupMessage);
+            notifyUser(notificationText, '{bold}{red-fg}PRIVATE KEYS COPIED TO CLIPBOARD{/}', 500);
+            settingsWindow.focus();
+        });
+    
+
+
+    ///////////////////////////////////////////////////////////////////
+    //  SYNC BAR
+    //////////////////////////////////////////////////////////////////
+    let progress = blessed.progressbar({
+        parent: walletWindow,
+        style: {
+            fg: 'red',
+            bg: 'default',
+            bar: {
+                bg: 'default',
+                fg: 'red'
+            },
+            border: {
+                fg: 'default',
+                bg: 'default'
+            }
+        },
+        ch: ':',
+        width: '100%',
+        height: 1,
+        top: '90%',
+        left: '0%',
+        filled: 0
+    });
+
+    // refresh the sync information every second
+    refreshSync(syncStatus, wallet, progress);
+    setInterval(refreshSync.bind(null, syncStatus, wallet, progress), 1000);
+
+    ///////////////////////////////////////////////////////////////////
+    //  TRANSFER WINDOW
+    //////////////////////////////////////////////////////////////////
+
+    // define error text
+    let errorText = blessed.text({
+        parent: transferWindow,
+        top: '75%',
+        left: 'center',
+        tags: true,
+        fg: 'red',
+    })
+
+    // create the opening menu
+    let transferForm = blessed.form({
+        parent: transferWindow,
+        keys: true,
+        left: 'center',
+        top: '5%',
+        width: 42,
+        height: 15,
+        bg: 'black',
+        fg: 'red',
+        border: {
+            type: 'line',
+            fg: 'white'
+        }
+    });
+
+    transferForm.on('click', function() {
+        screen.focusPop();
+    })
+
+    // transfer form post handling
+    transferForm.on('submit', async function(data) {
+        let paymentID;
```

Comment:
You never assign this correctly if a payment ID *is* given. Try this:
```suggestion
        const paymentID: string | undefined = data.paymentid === '' ? undefined : data.paymentid;
```